spec/requests/subjects_request_spec.rb

```diff
@@ -0,0 +1,120 @@
+require 'rails_helper'
+
+RSpec.describe 'Subjects', type: :request do
+  let(:subject) { create(:subject) }
+  let(:subjects) { create_list(:subject, 3) }
+
+  describe 'GET #index' do
+    before do
+      get subjects_path
+    end
+    it 'index status' do
+      expect(response).to have_http_status(:success)
+    end
+    it 'render index view' do
+      is_expected.to render_template :index
+    end
+  end
+
+  describe 'GET #show' do
+    before do
+      get subject_path(id: subject.id)
+    end
+    it 'show status' do
+      expect(response).to have_http_status(:success)
+    end
+    it 'should show subject' do
+      is_expected.to render_template :show
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    let!(:new_subject) { create :subject }
+    context 'delete from subjects table' do
+      before do
+        get subject_path(id: subject.id)
+      end
+      it 'delete subject' do
+        expect { new_subject.destroy }.to change { Subject.count }.by(-1)
+      end
+
+      it 'after delete response successful' do
+        expect(response).to be_successful
+      end
+
+      it 'render view new after destroy' do
+        is_expected.to render_template :show
+      end
+    end
+
+    context 'does not exist subject' do
+      it { expect { get '/subjects/99' }.to raise_error(ActiveRecord::RecordNotFound) }
+    end
+  end
+
+  describe 'GET #new' do
+    it 'create a new Subject' do
+      get new_subject_path
+      expect(assigns(:subject)).to be_a_new(Subject)
+    end
+    it 'renders the :new template' do
+      get new_subject_path
+      expect(response).to render_template :new
+    end
+  end
+
+  describe 'CREATE #create' do
+    context 'with valid data' do
+      before do
+        post subjects_path, params: { subject: { name: 'English' } }
+      end
+
+      it { expect(Subject.count).to eq(1) }
+      it 'redirects to the new subject' do
+        expect(response).to redirect_to Subject.last
+      end
+    end
+
+    context 'with invalid attributes' do
+      before do
+        post subjects_path, params: { subject: { name: '' } }
+      end
+
+      it 'does not save the new subject' do
+        expect { response }.to_not change(Subject, :count)
+      end
+      it 're-renders the new method' do
+        expect(response).to render_template :new
+      end
+    end
+  end
+
+  describe 'PATCH #update' do
+    let(:attr) { { name: 'Physics' } }
+    let(:wrong_attr) { { name: '' } }
+    before(:each) do
+      patch subject_path(id: subject.id), params: { id: subject.id, subject: attr }
+      subject.reload
+    end
+
+    context 'valid attributes' do
+      it { expect(response).to redirect_to(subject) }
+      it { expect(subject.name).to eq('Physics') }
+      it { expect(response).to be_redirect }
+    end
+
+    context 'with invalid attributes' do
+      before do
+        patch subject_path(id: subject.id), params: { id: subject.id, subject: wrong_attr }
```

Comment:
see in test logs that you have two calls here, with valid attributes and invalid ones

Reply:
How to see that it has 2 calls?